packages/next/lib/eslint/runLintCheck.ts

```diff
@@ -217,6 +217,10 @@ export async function runLintCheck(
       maxWarnings
     )
   } catch (err) {
+    // If the user decides to `.eslintignore` all of the files in the lint directory it should not throw an error during `next build` or `next lint`
+    if (err?.messageTemplate === 'all-files-ignored') {
+      return null
+    }
```

Comment:
This can be removed in place of disabling the `errorOnUnmatchedPattern` property (done here: https://github.com/vercel/next.js/pull/27119)